api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.7.x.xml

```diff
@@ -99,5 +99,14 @@
         	CREATE FUNCTION UUID() RETURNS UUID LANGUAGE SQL AS $$ SELECT uuid_generate_v1() $$;
         </sql>
     </changeSet>
+
+	<changeSet id="TRUNK-6208-2024-05-18-1004" author="k4pran" dbms="h2">
+		<!-- 
+		Changing the column type to int(10) due to Liquibase change where integer precision
+		greater than 10 is now interpreted as BIGINT in H2 databases. 
+		See: https://github.com/liquibase/liquibase/pull/4008/files 
+		-->
+		<modifyDataType tableName="program_attribute_type" columnName="retired_by" newDataType="int(10)"/>
```

Comment:
There are many columns of `type="INT"`
Is there a reason why you did this for only this?

Reply:
This was the only one that seemed to fail validation, but updated to include the others

Reply:
I think it was from running the integration tests that triggered the validation failure (before changes to this file)

Reply:
I'm not sure on this, let me take a look and see if I can figure out why

Reply:
@dkayiwa I think the reason it is failing validation only on changeset TRUNK-4755 might be because it is the first changeset that triggers 'createTable'  on the `program_attribute_type` table. 

If we take the column `location_tag_id` on table `location_tag` for example it is initially created on in `liquibase-schema-only-1.9.x.xml` without the display width

```
	<changeSet author="sunbiz" id="20110914-0306">
		<createTable tableName="location_tag">
			<column name="location_tag_id" type="int" autoIncrement="true">
				<constraints primaryKey="true" nullable="false"/>
			</column>
         ...
        </changeSet>
```

If I change this to int(11) it does fail for the same reason:

`org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [location_tag_id] in table [location_tag]; found [bigint (Types#BIGINT)], but expecting [integer (Types#INTEGER)]
`

Another point to mention is that the when there is another changeset to in `liquibase-update-to-latest-2.0.x.xml` that has a call to create `location_tag` it isn't actually running by default in this test because of the dbms:

```
	<changeSet id="20090224-1229" author="Keelhaul+bwolfe" dbms="mysql">
		<preConditions onFail="MARK_RAN">
			<not><tableExists tableName="location_tag"/></not>
		</preConditions>
		<createTable tableName="location_tag">
        ...
			<column name="creator" type="int(11)" >
				<constraints nullable="false" />
			</column>
        ...
	</changeSet>
```

But even if I remove this and it runs it doesn't fail validation unless I change the first create_table changeset to int(11). Found a similar pattern with another one I tried it with

`org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [concept_set_id] in table [concept_set]; found [bigint (Types#BIGINT)], but expecting [integer (Types#INTEGER)]`

I guess the first call to create_table sets up the schema and a mismatch in the data types later on do not have the same impact on validation but not sure if you want me to dig any deeper than this?

Reply:
No need to dig any deeper than this.
Since these are failures for h2 which we use only for tests, can we avoid having to add these changes to this file, even if are clearly marked with `dbms="h2"`, just for the sake of making this file cleaner? 😊

Reply:
yea sure, I rolled back the other other modifies except the one that causes the test failures

Reply:
Do you, by any chance, have an idea of how we can avoid having to add this changeset to the `liquibase-update-to-latest-2.7.x.xml` file? After all is it only for that failing test.

Reply:
I’m not sure because I guess we will still want to keep the test that does the hibernate validation, but I will look at other options to see if we can avoid adding the changeset

Reply:
> but I will look at other options to see if we can avoid adding the changeset

That will be awesome!

Reply:
I just changed the type from `int(11)` to `int` at https://github.com/openmrs/openmrs-core/pull/4637/commits/da88f5a9f27369d65b9f97549ccacef7c0c667aa just like the rest of the `retired_by` fields, then i removed the h2 specific changeset.